crates/uv-git/src/resolver.rs

```diff
@@ -38,6 +43,55 @@ impl GitResolver {
         self.0.get(reference)
     }
 
+    /// Resolve a Git URL to a specific commit without performing any Git operations.
+    ///
+    /// Returns a [`GitSha`] if the URL has already been resolved (i.e., is available in the cache),
+    /// or if it can be fetched via the GitHub API. Otherwise, returns `None`.
+    pub async fn github_fast_path(
+        &self,
+        url: &GitUrl,
+        client: ClientWithMiddleware,
+    ) -> Result<Option<GitSha>, GitResolverError> {
+        let reference = RepositoryReference::from(url);
+
+        // If we know the precise commit already, return it.
+        if let Some(precise) = self.get(&reference) {
+            return Ok(Some(*precise));
+        }
+
+        // If the URL is a GitHub URL, attempt to resolve it via the GitHub API.
+        let Some(GitHubRepository { owner, repo }) = GitHubRepository::parse(url.repository())
+        else {
+            return Ok(None);
+        };
+
+        // Determine the Git reference.
+        let rev = url.reference().as_rev();
+
+        let url = format!("https://api.github.com/repos/{owner}/{repo}/commits/{rev}");
+
+        debug!("Attempting GitHub fast path for: {url}");
+        let mut request = client.get(url);
+        request = request.header("Accept", "application/vnd.github.3.sha");
+        request = request.header("User-Agent", "uv");
```

Comment:
We should set a good example and add contact information and version too: `uv/{uv_version} (+https://github.com/astral-sh/uv)`

Reply:
I'll change this separately, since we already do this in `uv-git`.